tests/trainer/test_dataloaders.py

```diff
@@ -233,60 +233,135 @@ def test_dataloaders_passed_to_fn(tmpdir, ckpt_path, n):
     assert len(trainer.test_dataloaders) == n
 
 
+class DummyModel(BoringModel):
+
+    def training_step(self, batch, batch_idx):
+        self.log("loss", self.global_step)
+        return super().training_step(batch, batch_idx)
+
+    def validation_epoch_end(self, outputs):
+        self.log("val_log", self.current_epoch)
+
+
+class Counter(Callback):
+
+    def __init__(self):
+        super().__init__()
+        self.train_epoch_count = 0
+        self.val_epoch_count = 0
+        self.test_epoch_count = 0
+        self.train_batches_seen = 0
+        self.val_batches_seen = 0
+        self.test_batches_seen = 0
+
+    def on_train_batch_start(self, trainer, pl_module, batch, batch_idx, dataloader_idx):
+        if not trainer.sanity_checking:
+            self.train_batches_seen += 1
+
+    def on_train_epoch_start(self, trainer, pl_module):
+        if not trainer.sanity_checking:
```

Comment:
let's turn sanity check off in the test and remove this if? simpler logic and the counter counts all seen. in any case, sanity check would not call this hook